spec/processing/vips_spec.rb

```diff
@@ -0,0 +1,261 @@
+        convert << tiff.path
+      end
+
+      allow(instance).to receive(:file).and_return(CarrierWave::SanitizedFile.new(tiff.path))
+
+      instance.convert('png', 0)
+      expect(instance.file.extension).to eq('png')
+      expect(instance).to be_format('png')
+      expect(instance.file.size).not_to eq(0)
+    end
+  end
+
+  describe '#resize_to_fill' do
+    it "resizes the image to exactly the given dimensions and maintain file type" do
+      instance.resize_to_fill(200, 200)
+
+      expect(instance).to have_dimensions(200, 200)
+      expect(::Vips::Image.new_from_file(instance.current_path).get("vips-loader")).to match(/jpeg/)
+    end
+
+    it "resizes the image to exactly the given dimensions and maintain updated file type" do
+      instance.convert('png')
+      instance.resize_to_fill(200, 200)
+
+      expect(instance).to have_dimensions(200, 200)
+      expect(::Vips::Image.new_from_file(instance.current_path).get("vips-loader")).to match(/png/)
+      expect(instance.file.extension).to eq('png')
+    end
+
+    it "scales up the image if it smaller than the given dimensions" do
+      expect(::MiniMagick::Tool::Identify.new.verbose(instance.current_path).call).to_not include('Quality: 70')
+
+      instance.resize_to_fill(1000, 1000, combine_options: { saver: { quality: 70 } })
+
+      expect(instance).to have_dimensions(1000, 1000)
+      expect(::MiniMagick::Tool::Identify.new.verbose(instance.current_path).call).to include('Quality: 70')
+    end
+  end
+
+  describe '#resize_and_pad' do
+    it "resizes the image to exactly the given dimensions and maintain file type" do
+      instance.resize_and_pad(200, 200)
+
+      expect(instance).to have_dimensions(200, 200)
+      expect(::Vips::Image.new_from_file(instance.current_path).get("vips-loader")).to match(/jpeg/)
+    end
+
+    it "resizes the image to exactly the given dimensions and maintain updated file type" do
+      instance.convert('png')
+      instance.resize_and_pad(200, 200)
+
+      expect(instance).to have_dimensions(200, 200)
+      expect(::Vips::Image.new_from_file(instance.current_path).get("vips-loader")).to match(/png/)
+    end
+
+    it "scales up the image if it smaller than the given dimensions" do
+      instance.resize_and_pad(1000, 1000)
+
+      expect(instance).to have_dimensions(1000, 1000)
+    end
+
+    it "pads with black" do
+      instance.resize_and_pad(200, 200)
+
+      color_of_pixel(instance.current_path, 0, 0).tap do |color|
+        expect(color).to include('#000000')
+        expect(color).not_to include('none')
+      end
+    end
+
+    it "pads with transparent" do
+      instance.convert('png')
+      instance.resize_and_pad(200, 200, nil, 'centre', true)
+
+      expect(color_of_pixel(instance.current_path, 0, 0)).to include('none')
+    end
+
+    it "doesn't pad with transparent" do
+      binding.pry
+
+      instance.resize_and_pad(200, 200)
+      instance.convert('png')
```

Comment:
I could not make this spec pass(apart from `bindings.pry`), it always breaks with `VipsForeignLoad: images do not match` here.
Do you have any idea?

Reply:
I'm going to investigate a bit, as it's been a while since I touched this code :)